### PR TITLE
bears/hypertext: Add HTTPoliceLintBear.py

### PR DESCRIPTION
**Pull request by yash-nisar**
Closes https://github.com/coala/coala-bears/issues/596

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by yash-nisar** [comment]
CC @Makman2

File: bears/hypertext/HTTPoliceLintBear.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    REQUIREMENTS = {PipRequirement('HTTPolice', '0.5.2')}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_DETECT = {'Formatting', 'Syntax'}
+    SEE_MORE = 'https://github.com/vfaronov/httpolice'
+
+    severity_map = {'E': RESULT_SEVERITY.MAJOR,
+                    'C': RESULT_SEVERITY.NORMAL,
+                    'D': RESULT_SEVERITY.INFO}
```
Comment (Makman2) [inline bears/hypertext/HTTPoliceLintBear.py]: you have to provide this at `@linter`, not here^^

File: bears/hypertext/HTTPoliceLintBear.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    @staticmethod
+    def create_arguments(filename, file, config_file,
+                         silent: typed_list(str)=[]):
+        """
+        :param silent:
```
Comment (Makman2) [inline bears/hypertext/HTTPoliceLintBear.py]: hmm the name `silent` is very general, maybe you can make it more precise?

File: bears/hypertext/HTTPoliceLintBear.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            Silences the given list of notice IDs.
```
Comment (Makman2) [inline bears/hypertext/HTTPoliceLintBear.py]: is there a sensible default list?

File: tests/hypertext/httpolice_test_files/test_bad_header.har [hunk 4]
```diff
@@ -0,0 +1,37 @@
+{
+  "_warning": "This is not a real HAR file! It only contains the keys that are interesting to HTTPolice. Please do not use this as an example of a valid HAR file.",
```
Comment (Makman2) [inline tests/hypertext/httpolice_test_files/test_bad_header.har]: do you maybe have at least one real HAR file?

File: bears/hypertext/HTTPoliceLintBear.py (hunk 1 above)
Comment (yash-nisar) [inline bears/hypertext/HTTPoliceLintBear.py]: Ohh, I've seen this mapping outside the decorator in many bears. Nvm will raise an issue :wink:

File: bears/hypertext/HTTPoliceLintBear.py (hunk 3 above)
Comment (yash-nisar) [inline bears/hypertext/HTTPoliceLintBear.py]: Couldn't find one. Completely depends on the user IMO.

File: tests/hypertext/httpolice_test_files/test_bad_header.har (hunk 4 above)
Comment (yash-nisar) [inline tests/hypertext/httpolice_test_files/test_bad_header.har]: Yes, I have collected a real valid  HAR file manually, will update the tests.

Comment (yash-nisar) [inline tests/hypertext/httpolice_test_files/test_bad_header.har]: `test_valid_file` has a real `HAR` file :smile:

**Comment by yash-nisar** [comment]
@Makman2 Updated.

File: bears/hypertext/HTTPoliceLintBear.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    LICENSE = 'AGPL-3.0'
+    CAN_DETECT = {'Formatting', 'Syntax'}
+    SEE_MORE = 'https://github.com/vfaronov/httpolice'
+
+    @staticmethod
+    def create_arguments(filename, file, config_file,
+                         silence_ids: typed_list(str)=[]):
+        """
+        :param silence_ids:
+            Silences the given list of notice IDs.
```
Comment (Makman2) [inline bears/hypertext/HTTPoliceLintBear.py]: The documentation shall also state that these IDs are defined by HTTPolice, so people know where to lookup those :+1:

**Comment by yash-nisar** [comment]
Updated @Makman2

File: bears/hypertext/HTTPoliceLintBear.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    SEE_MORE = 'https://github.com/vfaronov/httpolice'
+
+    @staticmethod
+    def create_arguments(filename, file, config_file,
+                         httpolice_silence_ids: typed_list(str)=[]):
+        """
+        :param httpolice_silence_ids:
+            Silences the given list of notice IDs. You can get more information
+            about the available notices and their IDs at
+            https://httpolice.readthedocs.io/en/stable/notices.html.
```
Comment (Makman2) [inline bears/hypertext/HTTPoliceLintBear.py]: since it's RST, is a special syntax required for links?

Comment (Makman2) [inline bears/hypertext/HTTPoliceLintBear.py]: no it works fine :)

**Comment by Makman2** [comment]
ack 40dd5d4

**Comment by Makman2** [comment]
ack 40dd5d4

**Comment by Makman2** [comment]
@rultor merge

**Pull request merged**